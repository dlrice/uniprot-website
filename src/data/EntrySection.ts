enum EntrySectionType {
  Expression = 'Expression',
  FamilyAndDomains = 'FamilyAndDomains',
  Function = 'Function',
  Interaction = 'Interaction',
  NamesAndTaxonomy = 'NamesAndTaxonomy',
  PathologyOrBioTech = 'PathologyOrBioTech',
  ProteinProcessing = 'ProteinProcessing',
  Sequence = 'Sequence',
  Structure = 'Structure',
}

export default EntrySectionType;
