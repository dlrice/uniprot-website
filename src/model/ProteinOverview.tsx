import React, { Fragment } from 'react';
import { processProteinData, ProteinNamesData } from './ProteinNames';
import { Organism, OrganismData } from './Organism';
import { processGeneData, GeneNamesData } from './GeneNames';
import { Card, InfoList } from 'franklin-sites';

type ProteinOverviewDataProps = {
  data: {
    primaryAccession: string;
    uniProtId: string;
    proteinExistence: string;
    proteinDescription: ProteinNamesData['proteinDescription'];
    organism: OrganismData['organism'];
    genes: GeneNamesData['genes'];
  };
};

export const ProteinOverview: React.FC<ProteinOverviewDataProps> = ({
  data,
}) => {
  const { name: proteinName } = processProteinData(data);
  const { name: geneName } = processGeneData(data);
  const infoListData = [
    {
      title: 'Gene',
      content: geneName,
    },
    {
      title: 'Evidence',
      content: data.proteinExistence,
    },
  ];

  return (
    <Card
      title={
        <Fragment>
          {data.uniProtId} <small>{data.primaryAccession}</small>
        </Fragment>
      }
    >
      <h5>
        {proteinName} - <Organism data={data} />
      </h5>
      <InfoList infoData={infoListData} />
    </Card>
  );
};

export default ProteinOverview;
