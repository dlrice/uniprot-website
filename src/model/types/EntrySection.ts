enum EntrySectionType {
  Expression = 'Expression',
  FamilyAndDomains = 'FamilyAndDomains',
  Function = 'Function',
  Interaction = 'Interaction',
  Miscellaneous = 'Miscellaneous',
  NamesAndTaxonomy = 'NamesAndTaxonomy',
  PathologyOrBioTech = 'PathologyOrBioTech',
  ProteinProcessing = 'ProteinProcessing',
  Sequence = 'Sequence',
  Structure = 'Structure',
  SubCellularLocation = 'SubCellularLocation',
}

export default EntrySectionType;
