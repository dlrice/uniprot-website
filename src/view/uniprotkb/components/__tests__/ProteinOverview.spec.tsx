import React from 'react';
import { render } from '@testing-library/react';
import ProteinOverview, {
  annotationScoreToPercentage,
} from '../ProteinOverviewView';
import ProteinNamesUIDataJson from '../__mocks__/ProteinNamesUIData.json';
import EntrySection from '../../../../model/types/EntrySection';

describe('ProteinOverview component', () => {
  test('should render', () => {
    const transformedData = {
      proteinExistence: 'IT exists',
      annotationScore: 12.4,
    };
    transformedData[EntrySection.NamesAndTaxonomy] = ProteinNamesUIDataJson;

    const { asFragment } = render(
      <ProteinOverview transformedData={transformedData} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('annotationScoreToPercentage function', () => {
  /*
    0-19  | bin 1 | 20%
    20-39 | bin 2 | 40%
    40-59 | bin 3 | 60%
    60-79 | bin 4 | 80%
    >=80  | bin 5 | 100%
  */
  const annotationScoreToPercentageTests = [
    // [annotation score, percentage]
    [0, 20],
    [1, 20],
    [19, 20],
    [20, 40],
    [39, 40],
    [40, 60],
    [59, 60],
    [60, 80],
    [79, 80],
    [80, 100],
    [81, 100],
    [1000, 100],
  ];
  annotationScoreToPercentageTests.forEach(([annotationScore, percentage]) => {
    test(`Should convert annotationScore of ${annotationScore} to ${percentage}%`, () => {
      expect(annotationScoreToPercentage(annotationScore)).toEqual(percentage);
    });
  });
});
