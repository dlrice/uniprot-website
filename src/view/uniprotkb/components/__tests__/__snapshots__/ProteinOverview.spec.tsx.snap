// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProteinOverview component should render 1`] = `
<DocumentFragment>
  <div
    class="info-list"
  >
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        Evidence
      </div>
      <div
        class="info-list__item__content"
      >
        IT exists
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__item__title"
      >
        Annotation score
      </div>
      <div
        class="info-list__item__content"
      >
        <div
          class="doughnut-chart--medium colour-light-grey"
        >
          <div
            class="doughnut-chart--medium__left-wrap"
          >
            <div
              class="doughnut-chart--medium__left-wrap__loader colour-uniprot-blue"
              style="transform: rotate(0deg);"
            />
          </div>
          <div
            class="doughnut-chart--medium__right-wrap"
          >
            <div
              class="doughnut-chart--medium__right-wrap__loader colour-uniprot-blue"
              style="transform: rotate(72deg);"
            />
          </div>
          <div
            class="doughnut-chart--medium__inner-circle"
          >
            <span>
              20%
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
